pkg/manager/utils.go

```diff
@@ -20,12 +20,71 @@ import (
 	"context"
 	"fmt"
 
+	prometheusv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
 	"github.com/singularity-data/risingwave-operator/apis/risingwave/v1alpha1"
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type ServiceMonitorContextKey struct{}
+
+func ContextWithServiceMonitorFlag(ctx context.Context, useServiceMonitor bool) context.Context {
+	return context.WithValue(ctx, ServiceMonitorContextKey{}, useServiceMonitor)
+}
+
+func ServiceMonitorFlagFromContext(ctx context.Context) bool {
+	if v, ok := ctx.Value(ServiceMonitorContextKey{}).(bool); ok {
+		return v
+	}
+	return false
+}
+
+func GenerateServiceMonitor(serviceName string, portName string, rw *v1alpha1.RisingWave) *prometheusv1.ServiceMonitor {
+
+	var sm = prometheusv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-sm", serviceName),
```

Comment:
What is `%s-sm`? Also any example of collected metrics in Prometheus? (e.g. the series name of `job`, `instance`, etc.)

Reply:
short for `ServerMonitor` I guess. There is a strict naming convention for objects

Reply:
Yes, it's short for ServerMonitor.

Reply:
To avoid creation failure due to the too-long name...